src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -20,4 +24,31 @@ public LineResponse saveLine(LineRequest request) {
         Line persistLine = lineRepository.save(request.toLine());
         return LineResponse.of(persistLine);
     }
+
+    @Transactional()
+    public List<LineResponse> findAllLines() {
+        List<Line> lines = lineRepository.findAll();
+
+        return lines.stream()
+            .map(LineResponse::of)
+            .collect(Collectors.toList());
+    }
+
+    @Transactional(readOnly = true)
+    public LineResponse findId(Long id) {
+        return lineRepository.findById(id)
+            .map(LineResponse::of)
+            .orElseGet(LineResponse::new);
```

Comment:
이 부분 역시 존재하지 않는 노선을 찾으려 하면, status code가 200이면서 모든 필드의 값이 null인 응답이 내려가게 됩니다.
이 부분도 404 에러 등이 발생하도록 바꿔주는 편이 좋을 것 같습니다.
```suggestion
            .orElseThrow(() -> new RuntimeException("노선을 찾을 수 없습니다."));
```
와 같은 식으로 바꿔 쓸 수 있으니 참고 부탁드립니다.
(위 말씀드린 예시에서는 `RuntimeException`을 사용하였지만 권장되지 않으니 적절한 에러 클래스를 찾거나 커스텀으로 새로 만들어주시길 부탁드립니다.)

- 이건 꼭 고쳐주시길 부탁드립니다.

Reply:
감사합니다 ! `Optional`을 잘몰라서 대충 넘어갔었던 것 같습니다. 반성합니다. 😢 
피드백 감사합니다 !